File: app/src/main/java/com/meteoro/testeandroid/ui/contact/di/ContactComponent.java
package com.meteoro.testeandroid.ui.contact.di;

import com.meteoro.testeandroid.core.di.PerFragment;
import com.meteoro.testeandroid.core.di.component.LibraryComponent;
import com.meteoro.testeandroid.ui.contact.presentation.ContactFragment;

import dagger.Component;

@PerFragment
@Component(
        dependencies = {LibraryComponent.class},
        modules = {ContactModule.class}
)
public interface ContactComponent {
    void inject(ContactFragment fragment);
}

File: app/src/main/java/santander/com/br/invest/di/component/ContactComponent.kt
package santander.com.br.invest.di.component

import dagger.Component
import santander.com.br.invest.di.module.ContactModule
import santander.com.br.invest.di.module.RepositoryModule
import santander.com.br.invest.ui.fragments.ContactFragment
import javax.inject.Singleton

@Singleton
@Component(
    modules = [ContactModule::class, RepositoryModule::class]
)
interface ContactComponent {
  fun inject(fragment: ContactFragment)
}
